core/templates/pages/topics-and-skills-dashboard-page/modals/delete-skill-modal.component.ts

```diff
@@ -42,12 +44,43 @@ export class DeleteSkillModalComponent extends ConfirmOrCancelModal {
     super(ngbActiveModal);
   }
 
+  fetchTopicIdToDiagnosticTestSkillIds(
+      topicAssignments: AssignedSkill[]): void {
+    let allTopicIds = [];
+    for (let topic of topicAssignments) {
+      allTopicIds.push(topic.topicId);
+    }
+    let topicIdsNotEligibleForUnassignment: string[] = [];
+    this.topicsAndSkillsDashboardBackendApiService
+      .fetchTopicIdToDiagnosticTestSkillIdsAsync(allTopicIds).then(
+        (dict) => {
```

Comment:
Unclear what dict represents. It is too generic a variable name.

Reply:
Updated with `responseDict`.